packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -90,7 +91,11 @@ export const createAddressTransactionsProvider = (
           },
           retryBackoffConfig,
           tipBlockHeight$,
-          transactionsEquals
+          transactionsEquals,
+          // Do not re-fetch transactions twice on load when tipBlockHeight$ loads from storage first
+          // It should also help when using poor internet connection.
+          // Caveat is that local transactions might get out of date...
```

Comment:
What's the impact and resolution of this scenario?

Reply:
The issue is that provider operations are not cancelable, and this specific operation of fetching transactions can take a long time. Before this commit, when tip$ changed while it's still fetching transactions of a previous tip, it will run 2 operations in parallel which makes really bad performance given rate limiting. So basically there are 2 options to make it better:
a) Ignore tip updates until you finish fetching from a previous call - what this PR done. The impact is that the wallet could show that it's at Block n+1, but only have transactions from block n, and consider itself fully synced.
b) Queue the requests: guaranteed to be up-to-date when wallet "settles", but given poor internet connection it could never ever settle.

Reply:
Also, [f869472](https://github.com/input-output-hk/cardano-js-sdk/pull/204/commits/f86947273d4db80261b9e5fb2364e2922d75ced1) updated tip polling strategy to not refetch it until the wallet "settles" (all reqs resolved), OR times out using maxPollInterval, which is set to 100s by default. So the only happy path when this becomes an issue is on initial load, when it loads tip from storage.